### DISCUSSION TITLE
How to correctly add noise to an image

### DISCUSSION THREAD

**Discussion by eyaltemps** [dlemstra/Magick.NET#1188] [Help]
Hello,
I'm trying to add Noise to an MagickImage with the following command:
magicImage.AddNoise(NoiseType.Random, Channels.Default);
The resulting image is not valid, so the noise really ruined its visibility.
Is there a way to control the range of noise you add? What is the right way to do it?

Thank you!

**Comment by dlemstra** [comment]
Can you explain in more detail why it is not valid? And why are you using the `channels` overload?

**Reply by eyaltemps** [reply]
Hey, 
Attached is Zip file with 2 ICO files ("AWS_TwoPages" and "2_SinglePage") and their output after adding Random/Uniform Noise.
[IcoFilesWithNoise.zip](https://github.com/dlemstra/Magick.NET/files/8727503/IcoFilesWithNoise.zip)

I use the following code to Add noise for the single page image:
```
using (var image = new MagickImage(filePath, settings))
{
    image.AddNoise(NoiseType.Random);
    image.Write(IcoOutputPathSingle);
}
```
I use the following code to Add noise for multi page image:
```
using (var collection = new MagickImageCollection(filePath, settings))
  {
      foreach (MagickImage image in collection){
          image.AddNoise(NoiseType.Random);
      }
      collection.Write(IcoOutputPathMulti);
  }
```

After adding Random noise the ICO images aren't visible anymore (lots of colorful dots). The Uniform looks ok.
What is the correct way to add noise to an image, and ensure that the original visibility is not destroyed after adding the noise (I don't care the type of noise), should I always use Uniform?

Thanks

**Reply by dlemstra** [reply]
If you want to add noise but preserve the original image you should not use `NoiseType.Random`. This was added to replace the image with random pixels. On of the other options should be used in your situation.
